propolis/src/hw/nvme/bits.rs

```diff
@@ -0,0 +1,365 @@
+    /// Entry Latency
+    pub enlat: u32,
+    /// Exit Latency
+    pub exlat: u32,
+    /// Relative Read Throughput
+    pub rrt: u8,
+    /// Relative Read Latency
+    pub rrl: u8,
+    /// Relative Write Throughput
+    pub rwt: u8,
+    /// Relative Write Latency
+    pub rwl: u8,
+    /// Reserved
+    pub _resv: [u8; 16],
+}
+
+#[derive(Copy, Clone)]
+#[repr(C)]
+pub struct IdentifyController {
+    // bytes 0-255 - Controller Capabilities and Features
+    /// PCI Vendor ID
+    pub vid: u16,
+    /// PCI Subsystem Vendor ID
+    pub ssvid: u16,
+    /// Serial Number
+    pub sn: [u8; 20],
+    /// Model Number
+    pub mn: [u8; 40],
+    /// Firmware Revision
+    pub fr: [u8; 8],
+    /// Recommended Arbitration Burst
+    pub rab: u8,
+    /// IEEE OUI Identifier
+    pub ieee: [u8; 3],
+    /// Multi-Interface Capabilities
+    pub cmic: u8,
+    /// Maximum Data Transfer Size
+    pub mdts: u8,
+    /// Reserved
+    pub _resv1: [u8; 178],
+
+    // bytes 256-511 - Admin Command Set Attributes & Optional Controller Capabilities
+    /// Optional Admin Command Support
+    pub oacs: u16,
+    /// Abort Command Limit
+    pub acl: u8,
+    /// Asynchronous Event Request Limit
+    pub aerl: u8,
+    /// Firmware Updates
+    pub frmw: u8,
+    /// Log Page Attributes
+    pub lpa: u8,
+    /// Error Log Page Etnries
+    pub elpe: u8,
+    /// Number of Power States Support
+    pub npss: u8,
+    /// Admin Vedor Specific Command Configuration
+    pub avscc: u8,
+    /// Reserved
+    pub _resv2: [u8; 246],
+
+    // bytes 512-2047 - NVM Command Set Attributes
+    /// Submission Queue Entry Size
+    pub sqes: u8,
+    /// Completion Queue Entry Size
+    pub cqes: u8,
+    /// Reserved
+    pub _resv3: [u8; 2],
+    /// Number of Namespaces
+    pub nn: u32,
+    /// Option NVM Command Support
+    pub oncs: u16,
+    /// Fused Operation Support
+    pub fuses: u16,
+    /// Format NVM Attributes
+    pub fna: u8,
+    /// Volatile Write Cache
+    pub vwc: u8,
+    /// Atomic Write Unit Normal
+    pub awun: u16,
+    /// Atomic Write Unit Power Fail
+    pub awupf: u16,
+    /// NVM Vendor Specific Command Configuration
+    pub nvscc: u8,
+    /// Reserved
+    pub _resv4: [u8; 173],
+    /// Reserved (I/O Command Set Attributes)
+    pub _resv5: [u8; 1344],
+
+    // bytes 2048-3071 - Power State Descriptors
+    /// Power State Descriptors (PSD0-PSD31)
+    pub psd: [PowerStateDescriptor; 32],
+
+    // bytes 3072-4095 - Vendor Specific
+    /// Vendor Specific
+    pub vs: [u8; 1024],
+}
+
+// We can't derive Default since Default isn't impl'd
+// for [T; N] where N > 32 yet (rust #61415)
+impl Default for IdentifyController {
+    fn default() -> Self {
+        Self {
+            vid: 0,
+            ssvid: 0,
+            sn: [0; 20],
+            mn: [0; 40],
+            fr: [0; 8],
+            rab: 0,
+            ieee: [0; 3],
+            cmic: 0,
+            mdts: 0,
+            oacs: 0,
+            acl: 0,
```

Comment:
Maybe want a todo to track tying this back to whatever the actual implementation turns out to be. Same eventually with AERL.